[1.27.1] utils/generic/Mutex.h not getting staged

**Describe the bug**

When building app against Linux desktop build hits
```
.../out/debug/filament/include/utils/Mutex.h:23:10: fatal error: 'utils/generic/Mutex.h' file not found
#include <utils/generic/Mutex.h>
         ^~~~~~~~~~~~~~~~~~~~~~~
```

`utils/generic/Mutex.h` is missing from staged install.

**Workaround**

```
mkdir -p out/debug/filament/include/utils/generic
mkdir -p out/release/filament/include/utils/generic
cp libs/utils/include/utils/generic/Mutex.h out/debug/filament/include/utils/generic
cp libs/utils/include/utils/generic/Mutex.h out/release/filament/include/utils/generic
```

**To Reproduce**
Steps to reproduce the behavior:
1. On Linux host run
```
CC=/usr/lib64/ccache/clang CXX=/usr/lib64/ccache/clang++ CXXFLAGS="-stdlib=libc++ -fno-builtin" ./build.sh -i debug release
```

**Desktop (please complete the following information):**
 - OS: Linux - Ubuntu 20.4
 - Backend: Vulkan

--- a/libs/utils/CMakeLists.txt
+++ b/libs/utils/CMakeLists.txt
@@ -42,7 +42,8 @@ set(DIST_HDRS
         ${PUBLIC_HDR_DIR}/${TARGET}/unwindows.h
 )
 
-set(DIST_LINUX_HDRS
+# Use our custom mutex/condition only on ANDROID
+set(DIST_ANDROID_HDRS
         ${PUBLIC_HDR_DIR}/${TARGET_LINUX}/Mutex.h
 )
 
@@ -123,8 +124,8 @@ endif()
 set(INSTALL_TYPE ARCHIVE)
 install(TARGETS ${TARGET} ${INSTALL_TYPE} DESTINATION lib/${DIST_DIR})
 install(FILES ${DIST_HDRS} DESTINATION include/${TARGET})
-if (LINUX OR ANDROID)
-    install(FILES ${DIST_LINUX_HDRS} DESTINATION include/${TARGET_LINUX})
+if (ANDROID)
+    install(FILES ${DIST_ANDROID_HDRS} DESTINATION include/${TARGET_LINUX})
 else()
     install(FILES ${DIST_GENERIC_HDRS} DESTINATION include/${TARGET_GENERIC})
 endif()

